src/client.rs

```diff
@@ -2591,8 +2653,12 @@ mod test {
             .parse()
             .unwrap();
         let layer_data = vec![1u8, 2, 3, 4];
-        let layer_digest = sha256_digest(&layer_data);
-        c.push_blob(&layer_reference, &[1, 2, 3, 4], &layer_digest)
+        let layer = OciDescriptor {
```

Comment:
Ok, figured out why this was failing. The value returned from `push_blob` is the full URL, not just the sha:

```
layer digest: http://localhost:55038/v2/layer-repository/blobs/sha256:9f64a747e1b97f131fabb6b447296c9b6f0201e79fb3c5356e6c77e89b6a806a
layer digest struct: sha256:9f64a747e1b97f131fabb6b447296c9b6f0201e79fb3c5356e6c77e89b6a806a
thread 'client::test::test_mount' panicked at src/client.rs:2674:14:
Failed to mount: SpecViolationError("Expected HTTP Status 201 Created, got 202 Accepted instead")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

So in the `mount_blob` call beneath this, you'll need to do `&layer.digest` instead

Reply:
That makes sense, it's been fixed. Thanks for finding that

Reply:
It's solved now